src/main/java/uk/nickbdyer/httpserver/controllers/FileController.java

```diff
@@ -19,4 +24,40 @@ public Response get(Request request) {
         return new Response(200, new HashMap<>(), file);
     }
 
+    @Override
+    public Response patch(Request request) {
+        if (etagsMatch(request)) {
```

Comment:
I would extract the necessary data out of the request at this level and let the rest of the app not worry about "Request"